packages/build-utils/src/fs/glob.ts

```diff
@@ -45,16 +45,33 @@ export default async function glob(
   const dirsWithEntries = new Set<string>();
 
   for (const relativePath of files) {
-    const fsPath = normalizePath(path.join(options.cwd, relativePath));
+    const absPath = path.join(options.cwd, relativePath);
+    const fsPath = normalizePath(absPath);
+
     let stat = statCache[fsPath];
     assert(
       stat,
       `statCache does not contain value for ${relativePath} (resolved to ${fsPath})`
     );
+
     const isSymlink = symlinks[fsPath];
+
+    // When `follow` mode is enabled, ensure that the entry is not a symlink
+    // that points to outside of `cwd`
+    if (
+      options.follow &&
+      (isSymlink || (await lstat(fsPath)).isSymbolicLink())
+    ) {
+      const target = await readlink(absPath);
+      const absTarget = path.join(path.dirname(absPath), target);
+      if (path.posix.relative(options.cwd, absTarget).startsWith('../')) {
```

Comment:
I think there's a bug on Windows. No matter the link target, relative or absolute, `path.posix.relative()` will ALWAYS return a path that begins with `../` on Windows.

If you use `path.relative()`, then it will resolve the correct relative path, however on Windows, the path always starts with `..\\`.

```suggestion
      if (path.relative(options.cwd, absTarget).startsWith(`..${path.sep}`)) {
```